modules/sdk/src/test/scala/org/tessellation/sdk/infrastructure/snapshot/CurrencySnapshotCreatorSuite.scala

```diff
@@ -0,0 +1,28 @@
+package org.tessellation.sdk.infrastructure.snapshot
+
+import cats.effect.{IO, Resource}
+
+import org.tessellation.ext.kryo._
+import org.tessellation.kryo.KryoSerializer
+import org.tessellation.schema.generators.signatureProofGen
+import org.tessellation.sdk.cli.CliMethod.snapshotSizeConfig
+import org.tessellation.sdk.sdkKryoRegistrar
+
+import weaver.MutableIOSuite
+import weaver.scalacheck.Checkers
+
+object CurrencySnapshotCreatorSuite extends MutableIOSuite with Checkers {
+  type Res = KryoSerializer[IO]
+
+  def sharedResource: Resource[IO, KryoSerializer[IO]] =
+    KryoSerializer.forAsync[IO](sdkKryoRegistrar)
+
+  test("single signature size matches the size limit") { implicit kryo =>
+    forall(signatureProofGen) { proof =>
```

Comment:
I don't think you can really prove the size always matches if you are mocking the values. In reality the signature isn't always 142 chars long.

Reply:
Yeah. I would need to write an effectful generator to actually sign snapshots which isn't that trivial and needs some workarounds.

Reply:
https://stackoverflow.com/questions/52122705/java-ecdsawithsha256-signature-with-inconsistent-length

We use `val defaultSignFunc = "SHA512withECDSA"` which uses `ASN.1 DER` (variable length) encoding. I checked snapshots and signatures vary between `140` and `144` chars so for now I would suggest setting `144` and then trying `val defaultSignFunc = "SHA512withPLAIN-ECDSA"` taking backward compatibility into account.